src/compiler/sys.ts

```diff
@@ -16,7 +16,7 @@ namespace ts {
         createDirectory(path: string): void;
         getExecutingFilePath(): string;
         getCurrentDirectory(): string;
-        readDirectory(path: string, extension?: string, exclude?: string[]): string[];
+        readDirectory(path: string, extension?: string, exclude?: string[], depth?: number): string[];

```

Comment:
We've updated this on the VS host side as well.  I spoke to @mhegazy and @vladima -  we'll need to guard against this change in shims.ts:
 public readDirectory(rootDir: string, extension: string, exclude: string[], depth?: number): string[] {
            // Wrap the API changes for 1.8 release. This try/catch
            // should be removed once TypeScript 1.8 has shipped.
           let encoded: string;
            try {
                encoded = this.shimHost.readDirectory(rootDir, extension, JSON.stringify(exclude), depth);
            }
            catch (e) {
                encoded = this.shimHost.readDirectory(rootDir, extension, JSON.stringify(exclude));
            }
            return JSON.parse(encoded);
        }